File: backend/app/schemas/movie.py
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel


class MovieListItem(BaseModel):
    id: int
    title: str
    slug: str
    genre: str
    rating: str
    duration_min: int
    release_date: date
    poster_url: str
    backdrop_url: str
    status: str
    is_featured: bool

    model_config = {"from_attributes": True}


class MovieDetail(MovieListItem):
    trailer_url: str | None
    synopsis: str
    cast: str
    formats: list[str]


class ShowtimeCinema(BaseModel):
    id: int
    name: str
    location: str

    model_config = {"from_attributes": True}


class ShowtimeRoom(BaseModel):
    id: int
    name: str

    model_config = {"from_attributes": True}


class ShowtimeItem(BaseModel):
    id: int
    start_time: datetime
    end_time: datetime
    format: str
    language: str
    price: Decimal
    cinema: ShowtimeCinema
    room: ShowtimeRoom


class MovieShowtimeGroup(BaseModel):
    date: date
    items: list[ShowtimeItem]

File: frontend/src/types/movie.ts
export type MovieStatus = 'NOW_SHOWING' | 'COMING_SOON'
export type ShowtimeFormat = '2D' | '3D' | 'IMAX'

export interface MovieListItem {
  id: number
  title: string
  slug: string
  genre: string
  rating: string
  duration_min: number
  release_date: string
  poster_url: string
  backdrop_url: string
  status: MovieStatus
  is_featured: boolean
}

export interface MovieDetail extends MovieListItem {
  trailer_url: string | null
  synopsis: string
  cast: string
  formats: ShowtimeFormat[]
}

export interface ShowtimeCinema {
  id: number
  name: string
  location: string
}

export interface ShowtimeRoom {
  id: number
  name: string
}

export interface ShowtimeItem {
  id: number
  start_time: string
  end_time: string
  format: ShowtimeFormat
  language: string
  price: string
  cinema: ShowtimeCinema
  room: ShowtimeRoom
}

export interface MovieShowtimeGroup {
  date: string
  items: ShowtimeItem[]
}
